### PR TITLE
hw-7.1-solution

### PR DESCRIPTION
**Pull request by SviatoslavExpert**

### CODE REVIEW

File: src/main/java/core/basesyntax/ArrayList.java [hunk 1]
```diff
@@ -4,49 +4,111 @@
...
+    public ArrayList() {
+        this.data = new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(Object[] data, int size) {
+        this.data = data;
+        this.size = size;
+    }
+
+    public Object[] getData() {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/ArrayList.java]: do we need this method here?

File: src/main/java/core/basesyntax/ArrayList.java [hunk 2]
```diff
@@ -4,49 +4,111 @@
...
+        return data;
+    }
+
+    public void setData(Object[] data) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/ArrayList.java]: do we need this method here?

File: src/main/java/core/basesyntax/ArrayList.java [hunk 3]
```diff
@@ -4,49 +4,111 @@
...
+        this.data = data;
+    }
+
+    public int getSize() {
+        return size;
+    }
+
+    public void setSize(int size) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/ArrayList.java]: do we need this method here?

File: src/main/java/core/basesyntax/ArrayList.java [hunk 4]
```diff
@@ -4,49 +4,111 @@
...
+
+    public Object[] getData() {
+        return data;
+    }
+
+    public void setData(Object[] data) {
+        this.data = data;
+    }
+
+    public int getSize() {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/ArrayList.java]: do we need this method here?

File: src/main/java/core/basesyntax/ArrayList.java [hunk 5]
```diff
@@ -4,49 +4,111 @@
...
+        System.arraycopy(data, index + 1, data, index, size - 1);
+        data[size--] = null;
+        return result;
     }
 
     @Override
     public T remove(T t) {
-        return null;
+        for (int i = 0; i < data.length; i++) {
+            if (data[i] == t) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/ArrayList.java]:
```suggestion
            if (t! = null && data[i].equals(t)) {
```

File: src/main/java/core/basesyntax/ArrayList.java (hunk 1 above)
Comment (SviatoslavExpert) [reply src/main/java/core/basesyntax/ArrayList.java]: Fixed.

File: src/main/java/core/basesyntax/ArrayList.java (hunk 2 above)
Comment (SviatoslavExpert) [reply src/main/java/core/basesyntax/ArrayList.java]: Fixed.

File: src/main/java/core/basesyntax/ArrayList.java (hunk 3 above)
Comment (SviatoslavExpert) [reply src/main/java/core/basesyntax/ArrayList.java]: Fixed.

File: src/main/java/core/basesyntax/ArrayList.java (hunk 4 above)
Comment (SviatoslavExpert) [reply src/main/java/core/basesyntax/ArrayList.java]: Fixed.

File: src/main/java/core/basesyntax/ArrayList.java (hunk 5 above)
Comment (SviatoslavExpert) [reply src/main/java/core/basesyntax/ArrayList.java]: I tried to do this:  if (t! = null && data[i].equals(t)) , but it didn't pass the tests.  
So, I did this:  if (t! = null && data[i] == t).  
The tests went well.
Fixed.

Comment (SviatoslavExpert) [reply src/main/java/core/basesyntax/ArrayList.java]: Thank you.

File: src/main/java/core/basesyntax/ArrayList.java [hunk 6]
```diff
@@ -4,49 +4,95 @@
...
+        System.arraycopy(data, index + 1, data, index, size - 1);
+        data[size--] = null;
+        return result;
     }
 
     @Override
     public T remove(T t) {
-        return null;
+        for (int i = 0; i < data.length; i++) {
+            if (t != null && data[i] == t) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/ArrayList.java]:
```suggestion
            if (data[i].equals(t)) {
```

File: src/main/java/core/basesyntax/ArrayList.java [hunk 7]
```diff
@@ -4,49 +4,95 @@
...
 public class ArrayList<T> implements List<T> {
+    private Object[] data;
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+
+    public ArrayList() {
+        this.data = new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(Object[] data, int size) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/ArrayList.java]: do we need this constructor here?

**Pull request closed without merging**